process/block/metablock.go

```diff
@@ -1305,18 +1305,8 @@ func (mp *metaProcessor) isShardHeaderValidFinal(currHdr *block.Header, lastHdr
 
 // receivedShardHeader is a call back function which is called when a new header
 // is added in the headers pool
-func (mp *metaProcessor) receivedShardHeader(shardHeaderHash []byte) {
-	shardHeaderPool := mp.dataPool.Headers()
```

Comment:
You can keep lines between 1309-1312 and remove line 1368, exactly as you did in shardblock.go

Reply:
done.